datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -317,16 +317,27 @@ fn rename_expressions(
         .into_iter()
```

Comment:
As `rename_expressions` is getting more feature-full perhaps we can add some doc comments explaining what it is doing (I think it is basically renaming expressions from substrait to match DataFusion's expectations?)

Reply:
Sure, added some docs in https://github.com/apache/datafusion/pull/11049/commits/87829ece5f0473c001a56ecd6355b95436751fe1